test/e2e/specs/embedding.test.js

```diff
@@ -157,4 +166,43 @@ describe( 'Embedding content', () => {
 		await page.waitForSelector( 'input[value="https://twitter.com/thatbunty"]' );
 		await page.waitForSelector( 'input[value="https://wordpress.org/gutenberg/handbook/"]' );
 	} );
+
+	it( 'should handle non-embeddable pasted URLs', async () => {
+		await clickBlockAppender();
+		// URL that cannot be embedded.
+		await page.keyboard.type( 'https://twitter.com/wooyaygutenberg123454312' );
+		await pressWithModifier( META_KEY, 'a' ); // Select the URL we just typed.
+		await pressWithModifier( META_KEY, 'x' ); // Cut it.
+		await pressWithModifier( META_KEY, 'v' ); // Paste it. This should trigger the embed paste transformation.
+		// wait for a paragraph with the url in it
+		await page.waitForSelector( '.editor-block-list__block[data-type="core/paragraph"]' );
+		// Get the innerText from the rich text editor.
+		const text = await page.$eval(
+			'.editor-block-list__block[data-type="core/paragraph"] .editor-rich-text',
+			( element ) => element.innerText
+		);
+		expect( text ).toMatch( 'https://twitter.com/wooyaygutenberg123454312' );
+	} );
+
+	it( 'should handle embeddable pasted URLs', async () => {
+		await clickBlockAppender();
+		// URL that can be embedded.
+		await page.keyboard.type( 'https://www.youtube.com/watch?v=lXMskKTw3Bc' );
+		await pressWithModifier( META_KEY, 'a' ); // Select the URL we just typed.
+		await pressWithModifier( META_KEY, 'x' ); // Cut it.
+		await pressWithModifier( META_KEY, 'v' ); // Paste it. This should trigger the embed paste transformation.
```

Comment:
Does copy pasting work in puppeteer? Last time I checked it does nothing. So this test would still pass, but the nothing really happened?

Reply:
To ensure it works, let's check the content between cutting and pasting.

Reply:
Sure. I've verified it works visually with `PUPPETEER_HEADLESS=false` but an assertion here would be a good improvement.